tempto-core/src/main/java/io/prestosql/tempto/assertions/QueryResultValueComparator.java

```diff
@@ -261,6 +263,14 @@ private boolean timestampWithTimezoneEqual(Object actual, Object expected)
         return false;
     }
 
+    private boolean javaObjectEqual(Object actual, Object expected)
+    {
+        if (actual.getClass().equals(expected.getClass())) {
```

Comment:
Why? Wouldn't `actual.equals(expected)` be enough?

Reply:
Exactly, `actual.equals(expected)` is enough.